gio/src/subclass/action.rs

```diff
@@ -0,0 +1,417 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::sync::LazyLock;
+
+use glib::{
+    ffi::{gboolean, GVariant, GVariantType},
+    prelude::*,
+    subclass::prelude::*,
+    translate::*,
+};
+
+use crate::{ffi, Action};
+
+pub trait ActionImpl: ObjectImpl + ObjectSubclass<Type: IsA<Action>> {
+    fn name(&self) -> glib::GString {
+        self.parent_name()
+    }
+
+    fn parameter_type(&self) -> Option<glib::VariantType> {
+        self.parent_parameter_type()
+    }
+
+    fn state_type(&self) -> Option<glib::VariantType> {
+        self.parent_state_type()
+    }
+
+    fn state_hint(&self) -> Option<glib::Variant> {
+        self.parent_state_hint()
+    }
+
+    fn is_enabled(&self) -> bool {
+        self.parent_enabled()
+    }
+
+    fn state(&self) -> Option<glib::Variant> {
+        self.parent_state()
+    }
+
+    fn change_state(&self, value: glib::Variant) {
+        self.parent_change_state(value);
+    }
+
+    fn activate(&self, parameter: Option<glib::Variant>) {
+        self.parent_activate(parameter);
+    }
+}
+
+pub trait ActionImplExt: ActionImpl {
+    fn parent_name(&self) -> glib::GString {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Action>() as *const ffi::GActionInterface;
+
+            let func = (*parent_iface)
+                .get_name
+                .expect("no parent \"get_name\" implementation");
+            let ret = func(self.obj().unsafe_cast_ref::<Action>().to_glib_none().0);
+            from_glib_none(ret)
```

Comment:
This is rather suboptimal as it copies strings all the time whenever the name is retrieved. This should be possible to make a `&glib::GStr` or not?